### PR TITLE
CI: Add script to generate `bear-requirements.txt`

### PR DESCRIPTION
**Pull request by yukiisbored**
generate_requirements_bears.py is a script for generating a pip
requirements text file for linters that can be installed with pip

Closes https://github.com/coala/coala-bears/issues/1044

### CODE REVIEW

File: .ci/generate_requirements_bears.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+PROJECT_BEAR_DIR = os.path.abspath(os.path.join(PROJECT_DIR, 'bears'))
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description='This program generates a pip requirement file for '
+                    'installation of linters that are used by the bears.')
+    parser.add_argument('--output', '-o',
+                        help='output file for requirements',
```
Comment (yukiisbored) [inline .ci/generate_requirements_bears.py]: Does anyone have a better phrase for this?

Comment (jayvdb) [inline .ci/generate_requirements_bears.py]: 'name of file to generate, or - for stdout' (and add the stdout support)

**Comment by yukiisbored** [comment]
@jayvdb I've added `stdout` support

**Comment by yukiisbored** [comment]
`bear-requirements.txt`
```
autoflake~=0.6
autopep8~=1.2
bandit~=1.2
cmakelint~=1.3
cppclean~=0.12
cpplint~=1.3
eradicate~=0.1
guess-language-spirit~=0.5
html-linter~=0.3
isort~=4.2
language-check~=0.8
libclang-py3~=0.2
munkres3~=1.0
mypy-lang~=0.4
nbformat~=4.1
nltk~=3.1
proselint~=0.7
pycodestyle
pydocstyle~=1.1
pyflakes~=1.3.*
pylint~=1.6
pyyaml~=3.12
radon~=1.4
requests~=2.12
restructuredtext_lint~=0.17
rstcheck~=2.2
safety~=0.5
scspell3k~=2.0
vim-vint~=0.3
vulture~=0.10.0
yamllint~=1.5
yapf~=0.14
```

File: .ci/generate_requirements_bears.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description='This program generates a pip requirement file for '
+                    'installation of linters that are used by the bears.')
+    parser.add_argument('--output', '-o',
+                        help='name of file to generate, or - for stdout',
+                        default=os.path.join(PROJECT_DIR,
+                                             'bears-requirements.txt'))
```
Comment (jayvdb) [inline .ci/generate_requirements_bears.py]: can you change the default to `bear-requirements.txt` so it only has one plural in the filename.
And update the commit message to be the same.

File: .ci/generate_bear_requirements.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import sys
+import itertools
```
Comment (jayvdb) [inline .ci/generate_bear_requirements.py]: alpha order

File: .ci/generate_bear_requirements.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            if isinstance(requirement, PipRequirement) and \
+               requirement not in requirements:
+                requirements.append(requirement)
+
+    return sorted(requirements, key=lambda requirement: requirement.package)
+
+
+def write_requirements(requirements, output):
+    for requirement in requirements:
+        if requirement.version != '':
```
Comment (jayvdb) [inline .ci/generate_bear_requirements.py]: To allow other false-ish values, use
```
if requirement.version:
```

**Comment by jayvdb** [comment]
sorry, one more.  rename script from `generate_requirements_bears.py` to `generate_bear_requirements.py` , per name change made elsewhere. (and update commit message)

**Comment by jayvdb** [comment]
ack a998750

**Comment by jayvdb** [comment]
@rultor merge

**Comment by jayvdb** [comment]
ack 64f83f5

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**